x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -153,17 +137,13 @@ public Realms(Settings settings, Environment env, Map<String, Realm.Factory> fac
         if (licenseState.isAuthAllowed() == false) {
             return Collections.emptyList();
         }
-
-        AllowedRealmType allowedRealmType = licenseState.allowedRealmType();
-        switch (allowedRealmType) {
-            case ALL:
-                return Collections.unmodifiableList(realms);
-            case DEFAULT:
-                return Collections.unmodifiableList(standardRealmsOnly);
-            case NATIVE:
-                return Collections.unmodifiableList(nativeRealmsOnly);
-            default:
-                throw new IllegalStateException("authentication should not be enabled");
+        if (licenseState.areAllRealmsAllowed()) {
+            return realms;
+        } else if (licenseState.areStandardRealmsAllowed()) {
+            return standardRealmsOnly;
+        } else {
+            // native realms are basic licensed, and always allowed, even for an expired license
+            return nativeRealmsOnly;
```

Comment:
Continuation of the previous comment.

```suggestion
        if (licenseStateSnapshot.areAllRealmsAllowed()) {
            return realms;
        } else if (licenseStateSnapshot.areStandardRealmsAllowed()) {
            return standardRealmsOnly;
        } else {
            // native realms are basic licensed, and always allowed, even for an expired license
            return nativeRealmsOnly;
```